packages/rest/src/lib/RequestManager.ts

```diff
@@ -372,15 +372,28 @@ export class RequestManager extends EventEmitter {
 
 			// Attach all files to the request
 			for (const [index, file] of request.files.entries()) {
-				formData.append(file.key ?? `files[${index}]`, file.data, file.name);
+				const fileKey = file.key ?? `files[${index}]`;
+
+				// https://developer.mozilla.org/en-US/docs/Web/API/FormData/append#parameters
+				// FormData.append only accepts a string or Blob.
+				// https://developer.mozilla.org/en-US/docs/Web/API/Blob/Blob#parameters
+				// The Blob constructor accepts TypedArray/ArrayBuffer, strings, and Blobs.
+				if (typeof file.data === 'string') {
+					formData.append(fileKey, file.data, file.name);
+				} else if (Buffer.isBuffer(file.data)) {
```

Comment:
This won't actually match ArrayBuffers or TypedArrays, just nodejs buffers

Reply:
https://github.com/discordjs/discord.js/blob/402514ff323ccf1f8c95d295f044cf0bb5547c2e/packages/rest/src/lib/RequestManager.ts#L30

Reply:
I would *love* to enable TypedArray/ArrayBuffer/Blob/File, etc in the future though. Just not in this PR Lol

Reply:
Thats fair